core/src/main/java/org/kohsuke/stapler/Function.java

```diff
@@ -226,8 +222,9 @@ Object bindAndInvoke(Object o, StaplerRequest req, StaplerResponse rsp, Object..
             throw new AssertionError(e);    // impossible
         }
 
-        PARSE_METHODS = CacheBuilder.newBuilder().weakKeys().build(new CacheLoader<Class,Function>() {
-            public Function load(Class from) {
+        PARSE_METHODS = new ClassValue<Function>() {
+            @Override
+            public Function computeValue(Class<?> from) {
```

Comment:
Not only simpler, but should fix a memory leak in the rare case that `fromStapler` is actually defined: https://github.com/stapler/stapler/pull/213#discussion_r620275869